docs/pages/connect-your-client/vnet.mdx

```diff
@@ -0,0 +1,95 @@
+---
+title: Using VNet
+description: Using VNet
+---
+
+## How it works
+
+VNet is a virtual network emulation tool which enables any program on your computer to connect to
+TCP applications protected by Teleport, with no extra changes required to the program itself. It is
+available in Teleport Connect and tsh on macOS, with Windows and Linux version in the works.
+
+Let's say you work on software which needs to connect to a TCP-based application accessible
+through a Teleport cluster. Before VNet, you had two options. You could use `tsh app login` to
+obtain certs for that TCP app. But this meant that your software had to be aware of those certs and
+supply them to the TCP app. To avoid dealing with certs, you could set up an authenticated tunnel
+with `tsh proxy app --tunnel`. Your program could then connect to the tunnel, running on a random
+port on localhost, without worrying about certs. But now you had to manage the tunnel itself,
+remember to start it and remember what its port number is. This was quite inconvenient, especially
+when dealing with many TCP apps.
+
+VNet goes one step further. It automatically proxies connections on the fly whenever your
```

Comment:
I would start this section with a variation of this paragraph, since it describes the benefits of VNet without referring to earlier versions of Teleport. Maybe something like:

> VNet automatically proxies connections whenever your
computer wants to reach a TCP app available through Teleport. Your software can talk to
`tcp-app.teleport.example.com` instead of talking to a port on localhost. Underneath, VNet uses [the
same TLS routing technology](../architecture/tls-routing.mdx) as `tsh proxy app`, just with extra
DNS and virtual network setup on the client side.